plasmapy/simulation/particle_tracker.py

```diff
@@ -858,6 +858,9 @@ def _push(self) -> None:
             .T
         )
 
+        # Don't count untracked particles as being on the grid
+        self.on_grid[~tracked_mask] = False
```

Comment:
While testing the new stopping condition for proton radiography I discovered that this behavior wasn't already implemented